projects/kit/components/tree/components/tree-item/tree-item.component.ts

```diff
@@ -94,4 +94,8 @@ export class TuiTreeItem implements DoCheck {
     public checkChanges(): void {
         this.change$.next();
     }
+
+    public toggle(item: TuiTreeItem): void {
```

Comment:
Why there's a toggle method within TreeItem itself and you still need to pass the item to it? Also controller is something you already provide yourself, do we really need a toggle method? Developers already should have access to the controller whether because they are providing it or by `viewChild(TUI_TREE_CONTROLLER)`